packages/radix-ui-themes/src/helpers/props/layout.props.ts

```diff
@@ -136,10 +206,80 @@ function withLayoutProps(props: LayoutProps) {
     withBreakpoints(props.top, 'rt-r-top'),
     withBreakpoints(props.bottom, 'rt-r-bottom'),
     withBreakpoints(props.left, 'rt-r-left'),
-    withBreakpoints(props.right, 'rt-r-right'),
-  ]
-    .filter(Boolean)
-    .join(' ');
+    withBreakpoints(props.right, 'rt-r-right')
+  );
+}
+
+function getLayoutStyles(props: LayoutProps) {
+  const baseLayoutClassNamess = withLayoutProps(props);
+
+  const [gridColumnClassNames, gridColumnCustomProperties] = getResponsiveStyles({
+    allowAribtraryValues: true,
+    className: 'rt-r-gc',
+    customProperty: '--grid-column',
+    value: props.gridColumn,
+    values: layoutPropDefs.gridColumn.values,
+  });
```

Comment:
It's a bit lengthy to retrieve the classes and CSS vars like this, wondering if that’s fine or we should try to figure out a more succinct syntax for this stuff